File: services/api/python/routes/properties.py
"""
PROPERTY ANALYSIS ROUTES

AI-powered property analysis and lawn/garden feature detection.
"""

from fastapi import APIRouter, HTTPException
from pydantic import BaseModel
from typing import Optional, List
from core.ai_clients import get_openai_client
from core.database import get_supabase
import json

router = APIRouter()

class PropertyAnalysisRequest(BaseModel):
    property_id: str
    analysis_type: str = "full"  # full, lawn_only, garden_only

@router.post("/analyze")
async def analyze_property(request: PropertyAnalysisRequest):
    """
    AI-powered property analysis

    Placeholder for future computer vision integration.
    Currently provides text-based analysis and recommendations.
    """
    try:
        supabase = get_supabase()

        # Fetch property data
        property_response = supabase.table('properties').select('*').eq('id', request.property_id).single().execute()
        property_data = property_response.data

        if not property_data:
            raise HTTPException(status_code=404, detail="Property not found")

        # Build analysis prompt
        prompt = f"""
        Analyze this property and provide landscaping recommendations:

        Property Details:
        - Location: {property_data.get('city', '')}, {property_data.get('state', '')}
        - Lot Size: {property_data.get('lot_size_sqft', 'Unknown')} sq ft
        - Lawn Area: {property_data.get('lawn_area_sqft', 'Unknown')} sq ft
        - Garden Beds: {property_data.get('garden_beds_sqft', 'Unknown')} sq ft
        - Trees: {property_data.get('tree_count', 'Unknown')}

        Provide analysis in JSON format:
        {{
            "property_condition": "<excellent/good/fair/poor>",
            "recommended_services": ["<service 1>", "<service 2>"],
            "estimated_service_frequency": {{
                "mowing": "<frequency>",
                "fertilization": "<frequency>",
                "pruning": "<frequency>"
            }},
            "seasonal_recommendations": {{
                "spring": ["<task 1>", "<task 2>"],
                "summer": ["<task 1>", "<task 2>"],
                "fall": ["<task 1>", "<task 2>"],
                "winter": ["<task 1>", "<task 2>"]
            }},
            "estimated_monthly_cost": <number>,
            "special_considerations": ["<consideration 1>", "<consideration 2>"]
        }}
        """

        client = get_openai_client()
        response = client.chat.completions.create(
            model="gpt-4",
            messages=[
                {"role": "system", "content": "You are an expert landscaping consultant providing property analysis and service recommendations."},
                {"role": "user", "content": prompt}
            ],
            response_format={"type": "json_object"},
            temperature=0.3
        )

        analysis = json.loads(response.choices[0].message.content)

        # Store analysis results
        supabase.table('properties').update({
            'property_condition': analysis.get('property_condition'),
            'detected_features': analysis
        }).eq('id', request.property_id).execute()

        return {
            "property_id": request.property_id,
            "analysis": analysis,
            "model": "gpt-4"
        }

    except Exception as e:
        raise HTTPException(status_code=500, detail=f"Property analysis failed: {str(e)}")

@router.post("/estimate-lawn-area")
async def estimate_lawn_area(property_id: str):
    """
    Estimate lawn area from property data

    Placeholder for future satellite/aerial image analysis.
    """
    try:
        supabase = get_supabase()

        property_response = supabase.table('properties').select('*').eq('id', property_id).single().execute()
        property_data = property_response.data

        if not property_data:
            raise HTTPException(status_code=404, detail="Property not found")

        lot_size = property_data.get('lot_size_sqft', 0)
        building_area = property_data.get('building_area_sqft', 0)

        # Simple estimation (typically 40-60% of non-building area is lawn)
        non_building_area = lot_size - building_area
        estimated_lawn = non_building_area * 0.5

        return {
            "property_id": property_id,
            "lot_size_sqft": lot_size,
            "building_area_sqft": building_area,
            "estimated_lawn_area_sqft": round(estimated_lawn, 2),
            "note": "This is a basic estimation. For accurate measurements, consider aerial image analysis."
        }

    except Exception as e:
        raise HTTPException(status_code=500, detail=f"Lawn area estimation failed: {str(e)}")

File: services/api/src/routes/properties.js
/**
 * PROPERTY ROUTES
 *
 * API endpoints for property management, GIS data integration, and AI analysis.
 */

const express = require('express');
const router = express.Router();
const { authenticate } = require('../middleware/auth');

router.use(authenticate);

/**
 * @route   GET /api/v1/properties
 * @desc    List properties with optional filtering by neighborhood
 */
router.get('/', async (req, res) => {
  res.json({ message: 'List properties endpoint - to be implemented' });
});

/**
 * @route   POST /api/v1/properties
 * @desc    Create a new property
 */
router.post('/', async (req, res) => {
  res.json({ message: 'Create property endpoint - to be implemented' });
});

/**
 * @route   GET /api/v1/properties/:id
 * @desc    Get property details with GIS data
 */
router.get('/:id', async (req, res) => {
  res.json({ message: 'Get property endpoint - to be implemented' });
});

/**
 * @route   POST /api/v1/properties/:id/fetch-gis
 * @desc    Fetch GIS data from parcel APIs (lot boundaries, building footprint)
 */
router.post('/:id/fetch-gis', async (req, res) => {
  res.json({ message: 'Fetch GIS data endpoint - integrates with Google Maps/parcel APIs' });
});

/**
 * @route   POST /api/v1/properties/:id/analyze
 * @desc    AI-powered property analysis (lawn area estimation, feature detection)
 */
router.post('/:id/analyze', async (req, res) => {
  res.json({ message: 'AI property analysis endpoint - calls Python AI service' });
});

/**
 * @route   GET /api/v1/properties/nearby
 * @desc    Find properties near a location (for neighborhood clustering)
 */
router.get('/nearby', async (req, res) => {
  res.json({ message: 'Nearby properties endpoint - uses PostGIS spatial queries' });
});

module.exports = router;
